server/UserRegistration.java

```diff
@@ -0,0 +1,357 @@
+        do {
+            passwordChecks = true;
+            System.out.println(" ");
+            System.out.println("\nYou will now need to create a password for this account");
+            System.out.print("\nThe password must be (At least 8 letters long, contain a capital letter, a lowercase letter and numerical value)");
+            System.out.println("\nPlease create your new password: ");
+            newPassword = sc.nextLine();
+            if (!passwordLength(newPassword)) {
+                passwordChecks = false;
+            }
+            if (!passwordChecks(newPassword)) {
+                passwordChecks = false;
+            }
+        } while (!passwordChecks);
+        do {
+            System.out.println("\n Your new password is:" + " " + newPassword);
+            System.out.println("\nConfirm password (Y/N): ");
+            userAnswerPassword = sc.nextLine().toUpperCase();
+            if (userAnswerPassword.equals("N")) {
+                System.out.println("\nThe password must be (At least 8 letters long, contain a capital letter, a lowercase letter and numerical value)");
+                System.out.println("\nPlease create your new password: ");
+                newPassword = sc.nextLine();
+                System.out.println("Ok the password has now been updated:" + newPassword);
+                System.out.println("\nConfirm new password(Y/N): ");
+                userAnswerPassword = sc.nextLine().toUpperCase();
+            }
+        } while (!passwordChecks & userAnswerPassword.equals("N"));
+
+        System.out.println("\n Your new password is:" + " " + newPassword);
+        return newPassword;
+    }
+
+
+    public static boolean passwordLength(String newPassword) {
+        if (newPassword.length() > 7) {
+            return passwordChecks(newPassword);
+        } else {
+            System.out.println(" ");
+            System.out.println("\nSorry the password" + " " + newPassword + " " + "does not meet the requirements");
+            return false;
+        }
+    }
+
+    private static boolean passwordChecks(String newPassword) {
+        boolean numericalCheck = false;
+        boolean capitalCheck = false;
+        boolean lowerCaseCheck = false;
+        char c;
+        // For the password to be accepted all conditions must be true otherwise it will not be.
+        for (int i = 0; i < newPassword.length(); i++) {
+            c = newPassword.charAt(i);
+            if (Character.isDigit(c)) {
+                numericalCheck = true;
+            }
+            if (Character.isLowerCase(c)) {
+                lowerCaseCheck = true;
+            } else if (Character.isUpperCase(c)) {
+                capitalCheck = true;
+            }
+            if (numericalCheck && lowerCaseCheck && capitalCheck) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+
+    public static String getNewUserBalance() {
+        String userOpenBalance;
+        boolean isItAFloat;
```

Comment:
Nitpick: you don't need to be so specific, this can be renamed to simply say `isFloat`. This is a standard in naming the variables, and you'll see this way of naming variables in most libraries and frameworks